shard.lua

```diff
@@ -101,12 +105,6 @@ local function is_connected()
     return init_complete
 end
 
-local function wait_connection()
```

Comment:
Why do you delete it? You just inlined it in several places, but is looks like regression. The similar code is better to join into a function.